### PR TITLE
[tests] verify meshcop service after a `factoryreset`

### PR DESCRIPTION
**Pull request by superwhd**
Depends-On: openthread/ot-br-posix#1339

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py [hunk 1]
```diff
@@ -130,6 +130,12 @@ def test(self):
         self.check_meshcop_service(br1, host)
         self.check_meshcop_service(br2, host)
 
+        br1.factory_reset()
+        br1.set_network_name('ot-br-1-3')
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:134]: What's the purpose of setting a new network name here?

File: tests/scripts/thread-cert/node.py [hunk 2]
```diff
@@ -2159,6 +2159,21 @@ def reset(self):
         if self.is_otbr:
             self.set_log_level(5)
 
+    def factory_reset(self):
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/node.py:2162]: The code is almost the same as `reset` above.
Maybe we can call `reset` with argument:
e.g. 

```suggestion
    def factory_reset(self):
        self._reset(cmd='factoryreset')
```

File: tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py (hunk 1 above)
Comment (superwhd) [reply tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:134]: This test cases uses network name to distinguish the meshcop services. Here we set a new network name to ensure that the discovered meshcop services are those we expect.

Comment (simonlingoogle) [reply tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:134]: The BR should already have published the meshcop service before set_network_name. Would `set_network_name` cause it to re-publish?

Comment (superwhd) [reply tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:134]: Yes.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Pull request merged**